src/aspire/image/xform.py

```diff
@@ -319,6 +321,67 @@ def _forward(self, im, indices):
 
         return im
 
+    @abc.abstractproperty
+    def noise_var(self):
+        """
+        Concrete implementations are expected to provide a method that
+        returns the noise variance for the NoiseAdder.
+        """
+
+
+class CustomNoiseAdder(NoiseAdder):
+    """
+    Instantiates a NoiseAdder using the provided `noise_filter`.
+    """
+
+    @property
+    def noise_var(self):
+        """
+        Return noise variance.
+
+        CustomNoiseAdder will estimate noise_var by taking a sample of the noise.
+
+        If you require tuning the noise_var sampling, see `get_noise_var`.
+        """
+        return self.get_noise_var()
+
+    def get_noise_var(self, sample_n=100, sample_res=128):
```

Comment:
is `128` heuristically a good enough resolution to get a good variance estimate? Wondering why this as a default

Reply:
Good question.  I have not tried every possible input, but it seemed to work pretty well so far.  My expectation was that authors of specialized `NoiseAdder`s would take some responsibility for owning their noise variance calculation.  For Gaussian there are formulas (z or t) that can be used, but they depend on a confidence parameter anyway.

I added a note about this to the docstring.

Reply:
Cool thanks!